src/Elliptic/DiscontinuousGalerkin/SubdomainOperator/InitializeSubdomain.hpp

```diff
@@ -0,0 +1,521 @@
+      typename DataBox, typename... InboxTags, typename Metavariables,
+      typename ActionList, typename ParallelComponent,
+      Requires<tmpl::all<initialization_tags,
+                         tmpl::bind<db::tag_is_retrievable, tmpl::_1,
+                                    tmpl::pin<DataBox>>>::value> = nullptr>
+  static std::tuple<DataBox&&> apply(
+      DataBox& box, const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ElementId<Dim>& /*element_id*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    const auto& element = db::get<domain::Tags::Element<Dim>>(box);
+    for (const auto& [direction, neighbors] : element.neighbors()) {
+      const auto& orientation = neighbors.orientation();
+      const auto direction_from_neighbor = orientation(direction.opposite());
+      for (const auto& neighbor_id : neighbors) {
+        const LinearSolver::Schwarz::OverlapId<Dim> overlap_id{direction,
+                                                               neighbor_id};
+        // Background-agnostic geometry
+        using geometry_argument_tags =
+            tmpl::list<domain::Tags::InitialExtents<Dim>,
+                       domain::Tags::InitialRefinementLevels<Dim>,
+                       domain::Tags::Domain<Dim>,
+                       LinearSolver::Schwarz::Tags::MaxOverlap<OptionsGroup>>;
+        elliptic::util::mutate_apply_at<geometry_tags, geometry_argument_tags,
+                                        geometry_argument_tags>(
+            detail::initialize_overlap_geometry<Dim>, make_not_null(&box),
+            overlap_id, neighbor_id, direction_from_neighbor);
+        // Background fields
+        if constexpr (has_background_fields) {
+          initialize_background_fields(box, overlap_id);
+        }
+        // Normalize face normals
+        normalize_face_normals(box, overlap_id);
+      }  // neighbors in direction
+    }    // directions
+
+    return {std::move(box)};
+  }
+
+  template <
+      typename DataBox, typename... InboxTags, typename Metavariables,
+      typename ArrayIndex, typename ActionList, typename ParallelComponent,
+      Requires<not tmpl::all<initialization_tags,
+                             tmpl::bind<db::tag_is_retrievable, tmpl::_1,
+                                        tmpl::pin<DataBox>>>::value> = nullptr>
+  static std::tuple<DataBox&&> apply(
+      DataBox& /*box*/, const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::GlobalCache<Metavariables>& /*cache*/,
+      const ArrayIndex& /*array_index*/, ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    ERROR(
+        "Dependencies not fulfilled. Did you forget to terminate the phase "
+        "after removing options?");
+  }
+
+ private:
+  template <typename DbTagsList>
+  static void initialize_background_fields(
+      db::DataBox<DbTagsList>& box,
```

Comment:
pass by `not_null`

Reply:
Made this change in a fixup, but are you sure you prefer it? It means the patterns in `db::get` and db::mutate` are different to usual code.

Reply:
I think we should follow our code review guidelines :)